### ISSUE TITLE
Can't install MariaDB-client on Amazon Linux (ami-1ecae776)

### ISSUE THREAD

**Issue by nshenry03** [sous-chefs/mariadb#58]
I tried to use this cookbook to install mariadb::client on a new Amazon Linux server; however, it failed to install the MariaDB-client package.
### Steps To Reproduce

Spin up a new Amazon Linux Server... For Example:

``` bash
knife ec2 server create \
  --identity-file ~/.ssh/keys/id_rsa_ec2_us_east_1.pem \
  --security-group-ids 'sg-xxxxxxxx' \
  --ssh-user 'ec2-user' --image 'ami-1ecae776' \
  --subnet 'subnet-xxxxxxx' --flavor 't2.micro' \
  --region 'us-east-1' --availability-zone 'us-east-1a' \
  --node-name 'prod-us-east-1-mysqlclient01' \
  --tags Name='prod-us-east-1-mysqlclient01',Role='prod-mysqlclient' \
  --run-list 'recipe[mariadb::client]'' \
  --environment 'production' \
  --server-connect-attribute public_ip_address
```
### Expected Result

Server should finish installing w/o errors and a working MySQL Client installed.
### Actual Result

Chef client fails to run:

```
  * yum_package[MariaDB-client] action install
    * No candidate version available for MariaDB-client
    ================================================================================
    Error executing action `install` on resource 'yum_package[MariaDB-client]'
    ================================================================================

    Chef::Exceptions::Package
    -------------------------
    No candidate version available for MariaDB-client

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/mariadb/recipes/client.rb

     66:     package name
     67:   end

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/mariadb/recipes/client.rb:66:in `block in from_file'

    yum_package("MariaDB-client") do
      action :install
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      package_name "MariaDB-client"
      timeout 900
      flush_cache {:before=>false, :after=>false}
      declared_type :package
      cookbook_name "mariadb"
      recipe_name "client"
    end
```
### Cause

In https://github.com/sinfomicien/mariadb/blob/master/recipes/repository.rb :

```
    if node['platform'] == 'redhat'
      target_platform = "rhel#{node['platform_version'].to_i}"
    else
      target_platform = "#{node['platform']}#{node['platform_version'].to_i}"
    end
    yum_repository "mariadb-#{node['mariadb']['install']['version']}" do
      description 'MariaDB Official Repository'
      baseurl 'http://yum.mariadb.org/' + \
        node['mariadb']['install']['version'] + "/#{target_platform}-amd64"
      gpgkey 'https://yum.mariadb.org/RPM-GPG-KEY-MariaDB'
      action :create
    end
```

The 'baseurl' is wrong (it doesn't resolve to a valid MariaDB repo because 'target_platform' isn't valid)...
#### wrapper-mariadb/metadata.rb

```
...

depends          'mariadb'
```
#### wrapper-mariadb/recipes/default.rb

```
...
if node[:platform] == 'amazon'
  chef_gem 'chef-rewind'
  require 'chef/rewind'

  node.set[:mariadb][:use_default_repository] = true

  include_recipe 'mariadb::repository'

  rewind "yum_repository[mariadb-#{node[:mariadb][:install][:version]}]" do
    baseurl 'http://yum.mariadb.org/' + \
      node[:mariadb][:install][:version] + '/centos7-amd64'
  end
end
```
#### wrapper-mariadb/recipes/client.rb

```
...
%w{wrapper-mariadb::default mariadb::client}.each do |recipe|
  include_recipe recipe
end
```

Then just modify the run list: `--run-list 'recipe[wrapper-mariadb::client]''`

**Comment by sinfomicien** [comment]
Could propose a PR directly on the base code of the cookbook please (to adapt the platform) ?

**Comment by ghost** [comment]
This is also a problem on CentOS 7.2.  I have worked around it by setting

```
default['mariadb']['install']['type'] = false
```

and then installing the mariadb-server and mariadb packages directly in my wrapper cookbook.  This also taught me that the mysql2_chef_gem cookbook requires the mariadb-devel yum package.

**Comment by xorima** [comment]
Hi, 
Closing issue as we have released 2.0.0 which does not use recipes anymore, if you are still having an issue please re-open 

Thanks
